### PR TITLE
Remove "native filesystem" warning on Windows

### LINKED ISSUE #630: Failed to initialize native filesystem for Windows
Running detekt 1.0.0.RC5-6 on Windows gives bogus warnings:

        WARN: Failed to initialize native filesystem for Windows
        java.lang.RuntimeException: Could not find installation home path. Please make sure bin/idea.properties is present in the installation directory.
                at org.jetbrains.kotlin.com.intellij.openapi.application.PathManager.getHomePath(PathManager.java:97)
                at org.jetbrains.kotlin.com.intellij.openapi.application.PathManager.findBinFile(PathManager.java:173)
                at org.jetbrains.kotlin.com.intellij.util.lang.UrlClassLoader.loadPlatformLibrary(UrlClassLoader.java:320)
                at org.jetbrains.kotlin.com.intellij.openapi.util.io.win32.IdeaWin32.<clinit>(IdeaWin32.java:48)
                at org.jetbrains.kotlin.com.intellij.openapi.util.io.FileSystemUtil.getMediator(FileSystemUtil.java:77)
                at org.jetbrains.kotlin.com.intellij.openapi.util.io.FileSystemUtil.<clinit>(FileSystemUtil.java:69)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ZipHandler.setFileStampAndLength(ZipHandler.java:69)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ZipHandler$1.createAccessor(ZipHandler.java:51)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ZipHandler$1.createAccessor(ZipHandler.java:47)
                at org.jetbrains.kotlin.com.intellij.util.io.FileAccessorCache.createHandle(FileAccessorCache.java:66)
                at org.jetbrains.kotlin.com.intellij.util.io.FileAccessorCache.get(FileAccessorCache.java:52)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ZipHandler.getCachedZipFileHandle(ZipHandler.java:112)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ZipHandler.createEntriesMap(ZipHandler.java:87)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.ArchiveHandler.getEntriesMap(ArchiveHandler.java:190)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.jar.CoreJarHandler.<init>(CoreJarHandler.java:42)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.jar.CoreJarFileSystem$1.create(CoreJarFileSystem.java:37)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.jar.CoreJarFileSystem$1.create(CoreJarFileSystem.java:33)
                at org.jetbrains.kotlin.com.intellij.util.containers.FactoryMap.get(FactoryMap.java:58)
                at org.jetbrains.kotlin.com.intellij.openapi.vfs.impl.jar.CoreJarFileSystem.findFileByPath(CoreJarFileSystem.java:50)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.findJarRoot(KotlinCoreEnvironment.kt:352)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.contentRootToVirtualFile(KotlinCoreEnvironment.kt:333)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.access$contentRootToVirtualFile(KotlinCoreEnvironment.kt:114)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$4.invoke(KotlinCoreEnvironment.kt:216)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$4.invoke(KotlinCoreEnvironment.kt:114)
                at org.jetbrains.kotlin.cli.jvm.compiler.ClasspathRootsResolver.convertClasspathRoots(ClasspathRootsResolver.kt:70)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:223)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment.<init>(KotlinCoreEnvironment.kt:114)
                at org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment$Companion.createForProduction(KotlinCoreEnvironment.kt:401)
                at io.gitlab.arturbosch.detekt.api.ProjectExtensionKt.<clinit>(ProjectExtension.kt:20)
                at io.gitlab.arturbosch.detekt.core.KtCompiler.<init>(KtCompiler.kt:17)
                at io.gitlab.arturbosch.detekt.core.KtTreeCompiler.<init>(KtTreeCompiler.kt:15)
                at io.gitlab.arturbosch.detekt.core.KtTreeCompiler$Companion.instance(KtTreeCompiler.kt:23)
                at io.gitlab.arturbosch.detekt.core.Detektor.run$default(Detektor.kt:25)
                at io.gitlab.arturbosch.detekt.cli.Runner.execute(Runner.kt:20)
                at io.gitlab.arturbosch.detekt.cli.Main.main(Main.kt:19)

The task still succeeds though (if no violations are found, of course).

### PR DESCRIPTION
**Pull request by 3flex**
Fixes this error when running Detekt:

WARN: Failed to initialize native filesystem for Windows
    java.lang.RuntimeException: Could not find installation home path. Please make sure bin/idea.properties is present in the installation directory.

Fixes #630

### CODE REVIEW

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt [hunk 1]
```diff
@@ -17,13 +18,18 @@ import org.jetbrains.kotlin.config.CompilerConfiguration
...
 
-val psiProject = KotlinCoreEnvironment.createForProduction(Disposer.newDisposable(),
-		CompilerConfiguration(), EnvironmentConfigFiles.JVM_CONFIG_FILES).project.apply {
-	makeMutable(this as MockProject)
-}
+val psiProject = createKotlinCoreEnvironment()
 
 val psiFactory = KtPsiFactory(psiProject, false)
 
+private fun createKotlinCoreEnvironment(): Project {
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt]: Maybe we can add a comment for clarity here.

File: detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt [hunk 2]
```diff
@@ -17,13 +18,18 @@ import org.jetbrains.kotlin.config.CompilerConfiguration
...
+	System.setProperty("idea.io.use.fallback", "true")
```
Comment (schalkms) [inline detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt]: Interesting setting. Could you please explain the behavior of it?

Comment (3flex) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt]: See https://github.com/arturbosch/detekt/issues/630#issuecomment-353222933 for background.

Comment (sschuberth) [reply detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/ProjectExtension.kt]: While it's good to have the explanation in GitHub, it should really also go to the code as a comment, IMO.

**Pull request merged**